### PR TITLE
Handle unittest.SkipTest exception with non-ascii characters

### LINKED ISSUE #4669: pytest  dose not support raise unittest Skiptest in chinese
- [pip list ] 
Package               Version
--------------------- -------
asn1crypto            0.24.0
atomicwrites          1.2.1
attrs                 18.2.0
bcrypt                3.1.5
cffi                  1.11.5
colorama              0.4.0
cryptography          2.4.2
enum34                1.1.6
funcsigs              1.0.2
idna                  2.8
ipaddress             1.0.22
lxml                  4.2.5
more-itertools        4.3.0
namedlist             1.7
nose                  1.3.7
nose-allure-plugin    1.0.5
nose-printlog         0.2.0
noselog               0.0.5
paramiko              2.4.2
pathlib2              2.3.2
Pillow                5.3.0
pip                   18.1
pluggy                0.8.0
py                    1.7.0
pyasn1                0.4.4
pycparser             2.19
PyMySQL               0.9.3
PyNaCl                1.3.0
pytest                3.9.3
pytest-allure-adaptor 1.7.10
pytest-ordering       0.6
scandir               1.9.0
selenium              2.53.3
setuptools            40.5.0
six                   1.11.0
wheel                 0.32.2

- [ ] pytest and operating system versions
pytest
(venv) E:\workspace\nosetest_lear\test_case>pytest --version
This is pytest version 3.9.3, imported from e:\workspace\nosetest_lear\venv\lib\site-packages\pytest.pyc
setuptools registered plugins:
  pytest-ordering-0.6 at e:\workspace\nosetest_lear\venv\lib\site-packages\pytest_ordering\__init__.pyc
  pytest-allure-adaptor-1.7.10 at e:\workspace\nosetest_lear\venv\lib\site-packages\allure\pytest_plugin.py

system:
win7

python version:
Python 2.7.8

- [ ] Minimal example if possible

```python
import unittest
class TestClass():
    def test_io(self):
        raise unittest.SkipTest(u'跳过')
```

when pytest run, the error is 
qq 20190123180512

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #4669

### CODE REVIEW

File: testing/test_nose.py [hunk 1]
```diff
@@ -366,3 +367,17 @@ def test_method(self):
...
+
+
+def test_skip_test_with_unicode(testdir):
+    testdir.makepyfile(
+        """
+        # encoding: utf-8
+        import unittest
+        class TestClass():
+            def test_io(self):
+                raise unittest.SkipTest(u'😊')
```
Comment (RonnyPfannschmidt) [inline testing/test_nose.py]: how would this fare against ` unittest.SkipTest(b'\0x01')

Comment (nicoddemus) [reply testing/test_nose.py]: Works OK I guess:

```
.tmp\test-skip-chinese.py s                                    [100%]
====================== short test summary info ======================
SKIPPED [1] c:\users\bruno\pytest\src\_pytest\nose.py:30:  x01
```

Comment (asottile) [reply testing/test_nose.py]: probably fails in python2.x with `b'\x21\xef\x1a'` (I picked some haphazard bytes that I assume aren't valid ascii) -- I assume @RonnyPfannschmidt meant to pick some non-ascii bytes? `b'\0x01` is a null byte followed by literal "x01"

Comment (RonnyPfannschmidt) [reply testing/test_nose.py]: @asottile yikes, good catch, in my experimentation things seemed fine tho

**Pull request merged**